File: bench/src/fib.js
// bench kernel: naive recursive fib(32) — fixed input, deterministic.
// Stresses call overhead, argument passing, and integer arithmetic.
"use strict";
function fib(n) {
  if (n < 2) return n;
  return fib(n - 1) + fib(n - 2);
}
console.log(`fib(32)=${fib(32)}`);

File: bench/src/fib.c
// bench kernel: naive recursive fib(32) — fixed input, deterministic.
// Stresses call overhead, argument passing, and integer arithmetic.
#include <stdio.h>

static long long fib(long long n) {
  if (n < 2)
    return n;
  return fib(n - 1) + fib(n - 2);
}

int main(void) {
  printf("fib(32)=%lld\n", fib(32));
  return 0;
}
